x/finality/keeper/power_dist_change.go

```diff
@@ -95,42 +96,57 @@ func (k Keeper) HandleFPStateUpdates(ctx context.Context, prevDc, newDc *ftypes.
 
 	newlyActiveFPs := newDc.FindNewActiveFinalityProviders(prevDc)
 	for _, fp := range newlyActiveFPs {
+		// what was the status prior to update the status?
+		prevStatus := fp.Status()
```

Comment:
@GAtom22 what do you think about storing the previous state in FinalityProviderDistInfo and every time we emit an event also update there :eyes: